### PR TITLE
Add localization infrastructure with FindBar pilot

### PR DESCRIPTION
**Pull request by jschick04**
## Summary

Adds localization infrastructure (IStringLocalizer + RESX + locale-aware formatting + RTL direction resolution) that **follows the OS culture with no in-app language setting**, plus a single pilot surface (FindBar) migrated end-to-end to prove the wiring. Bulk string migration is deferred.

Scope decided up front: **infra + one pilot surface**, follow the OS culture (no picker), bulk migration deferred.

## Infrastructure
- `Microsoft.Extensions.Localization` 10.0.10 + an `AddEventLogLocalization()` DI extension.
- `SharedResource` marker at the root `EventLogExpert.UI` namespace with `ResourcesPath="Resources"`, so the manifest resolves to `EventLogExpert.UI.Resources.SharedResource` (avoids the double-`Resources` segment footgun).
- Neutral English `SharedResource.resx` + a `qps-ploc` pseudo-loc canary satellite (dedicated `Canary_*` keys, always embedded).
- `<NeutralLanguage>en</NeutralLanguage>` for correct satellite / main-assembly behavior.
- `ContentCulture` resolver: `SupportedUiCultures` (`{en}`), a parent-walk `Resolve`, and `DirectionOf` (ltr/rtl) for future RTL.
- `DocumentInitScript.Build(...)` — an extracted, testable WebView document-init script builder (theme + dir + lang) replacing the inline string in `MainPage.xaml.cs`.

## FindBar pilot
- All FindBar user-facing strings → `@Localizer[...]`.
- Match count uses culture-aware `:N0` formatting.
- Wrap announcement string → a `FindWrapState` enum (`None` / `WrappedToFirst` / `WrappedToLast`), localized at render.

## Tests (UI.Tests 1556 → 1558)
- Guard/config, resolver edge cases (invariant / en-GB / zh-Hans / ar-SA), a satellite drift guard, and a no-hardcoded-pin source scan.
- Culture-sensitive FindBar tests (qps-Ploc canary + de-DE count formatting) with a culture-restoring bUnit context and a non-parallel collection.
- `DocumentInitScript` pin tests (null-theme + with-theme).

## Regression risk / follow-ups
- Every OS culture resolves to `en` / `ltr` today (`SupportedUiCultures = {en}`), so runtime behavior is identical to the previous static `<html lang="en">` — **zero v1 regression risk**.
- RTL robustness (mutation-observer / awaiting doc-created) and any `CurrentUICulture` OS-display-language bootstrap are documented follow-ups, needed only before shipping a non-English locale.

### CODE REVIEW

File: src/EventLogExpert.UI/Common/Interop/DocumentInitScript.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        ArgumentNullException.ThrowIfNull(language);
+
+        var themeStatement = themeAttribute is null ?
+            "root.removeAttribute('data-theme');" :
+            $"root.setAttribute('data-theme','{themeAttribute}');";
+
+        return
+            "(function(){var root=document.documentElement;if(!root){return;}" +
+            themeStatement +
+            $"root.setAttribute('dir','{direction}');root.setAttribute('lang','{language}');" +
```
[the remark this answers is not in the document]
Comment (jschick04) [reply src/EventLogExpert.UI/Common/Interop/DocumentInitScript.cs]: Addressed in ead812c5. The document-created script now wraps the work in an `apply()` that runs immediately when `document.documentElement` is present, and otherwise defers to `DOMContentLoaded`:

```js
(function(){function apply(){var root=document.documentElement;if(!root){return false;}/* set data-theme/dir/lang */return true;}if(!apply()){document.addEventListener('DOMContentLoaded',apply);}})();
```

The deferral cannot miss the event: a null `documentElement` means the `<html>` start tag hasn't been processed, so `DOMContentLoaded` (dispatched at end-of-parse) has not fired yet. `apply()` therefore runs exactly once in both paths, applying theme/dir/lang. The exact-script pin tests were updated to match. Thanks!

**Review by NikTilton — APPROVED** [review]
